### PR TITLE
Fission MQT integration with keda

### PR DESCRIPTION
**Pull request by therahulbhati**
Integrating KEDA into fission mqt to leverage benefits of KEDA component.

---
This change is [Reviewable](https://reviewable.io/reviews/fission/fission/1657)

### CODE REVIEW

**Comment by therahulbhati** [comment]
HPA issue with GKE 1.16 as metrics provider does not work due to it requiring a specific name for the cluster role. https://github.com/kedacore/keda/issues/515

**Comment by therahulbhati** [comment]
HPA issue in GKE 1.16, metrics provider does not work due to it requiring a specific name for the cluster role https://github.com/kedacore/keda/issues/515

File: pkg/mqtrigger/scalermanager.go [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": mqt.ObjectMeta.Name,
+					},
+				},
+				Spec: apiv1.PodSpec{
+					Containers: []apiv1.Container{
+						{
+							Name:            mqt.ObjectMeta.Name,
+							Image:           "rahulbhati/test:10",
```
Comment (vishal-biyani) [inline pkg/mqtrigger/scalermanager.go:419]: @therahulbhati this should come from values.yaml - so for example for Kafka we have something like https://github.com/fission/fission/blob/master/charts/fission-all/values.yaml#L179 - similarly  we can have a new structure something like:

```
mq_keda:
  connector_images:
    kafka: fission/kafka-keda-connector:1.10.0
```

File: test/tests/mqtrigger_v2/kafka/hellokafka.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+module.exports = async function (context) {
```
Comment (vishal-biyani) [inline test/tests/mqtrigger_v2/kafka/hellokafka.js:1]: nitpick: rename directories from mqtrigger_v2 to mqtrigger_kind_keda

Comment (therahulbhati) [reply test/tests/mqtrigger_v2/kafka/hellokafka.js]: Done

File: pkg/mqtrigger/scalermanager.go (hunk 1 above)
Comment (therahulbhati) [reply pkg/mqtrigger/scalermanager.go]: Done

File: charts/fission-all/values.yaml [hunk 3]
```diff
@@ -300,3 +300,9 @@ canaryDeployment:
 # Use these flags to enable opentracing, the variable is endpoint of Jaeger collector in the format shown below
 #traceCollectorEndpoint: "http://jaeger-collector.jaeger.svc:14268/api/traces?format=jaeger.thrift"
 #traceSamplingRate: 0.75
+
+## Message Queue Trigger Kind, KEDA: enable and configuration
+mqt_keda:
+  enabled: true
+  connector_images:
+    kafka: rahulbhati/test:10
```
Comment (vishal-biyani) [inline charts/fission-all/values.yaml:308]: The image should be `fission/keda-kafka:110`

Comment (therahulbhati) [reply charts/fission-all/values.yaml:308]: Done

File: charts/fission-all/templates/deployment.yaml [hunk 4]
```diff
@@ -839,3 +839,46 @@ spec:
...
+
+{{- if .Values.mqt_keda.enabled }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: mqtrigger-keda
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    svc: mqtrigger
```
[the remark this answers is not in the document]
Comment (therahulbhati) [reply charts/fission-all/templates/deployment.yaml]: Done

**Comment by vishal-biyani** [comment]
Thanks a lot for this one @therahulbhati - great job on getting this one done. 🎉

**Pull request merged**